### PR TITLE
triton-client: add triton-client http bindings for python

### PR DESCRIPTION
**Pull request by roger652**
@ichergui Here is the master PR for the triton-client python bindings.

The changes are the same as in the `dunfell` branch except:
- the obvious adjustments for the new recipe syntax
- fixes for compiler issues in the `triton-python-backend` which were not present in `dunfell`.

### CODE REVIEW

File: recipes-devtools/triton/triton-client/0001-fix-cmake-build.patch [hunk 1]
```diff
@@ -108,6 +133,19 @@ index a641fcb..dc13fa3 100644
...
+diff --git a/src/c++/examples/CMakeLists.txt b/src/c++/examples/CMakeLists.txt
+index c1fe4f9..bc0feb8 100644
+--- a/src/c++/examples/CMakeLists.txt
++++ b/src/c++/examples/CMakeLists.txt
+@@ -24,7 +24,7 @@
+ # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ 
+-cmake_minimum_required (VERSION 3.18)
++cmake_minimum_required (VERSION 3.16)
```
Comment (ichergui) [inline recipes-devtools/triton/triton-client/0001-fix-cmake-build.patch:145]: I don't think that is needed to change the `cmake_minimum_required` here

File: recipes-devtools/triton/triton-client/0001-fix-cmake-build.patch [hunk 2]
```diff
@@ -175,6 +213,245 @@ index e0eed14..ad47a7b 100644
...
+diff --git a/src/c++/perf_analyzer/CMakeLists.txt b/src/c++/perf_analyzer/CMakeLists.txt
+index 3682d32..65c30f8 100644
+--- a/src/c++/perf_analyzer/CMakeLists.txt
++++ b/src/c++/perf_analyzer/CMakeLists.txt
+@@ -24,7 +24,7 @@
+ # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ 
+-cmake_minimum_required (VERSION 3.18)
++cmake_minimum_required (VERSION 3.16)
```
Comment (ichergui) [inline recipes-devtools/triton/triton-client/0001-fix-cmake-build.patch:225]: Same as above

File: recipes-devtools/triton/triton-client/0001-fix-cmake-build.patch [hunk 3]
```diff
@@ -175,6 +213,245 @@ index e0eed14..ad47a7b 100644
...
+diff --git a/src/c++/perf_analyzer/client_backend/CMakeLists.txt b/src/c++/perf_analyzer/client_backend/CMakeLists.txt
+index 23da6f3..d4d78ce 100644
+--- a/src/c++/perf_analyzer/client_backend/CMakeLists.txt
++++ b/src/c++/perf_analyzer/client_backend/CMakeLists.txt
+@@ -24,7 +24,7 @@
+ # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ 
+-cmake_minimum_required (VERSION 3.18)
++cmake_minimum_required (VERSION 3.16)
```
Comment (ichergui) [inline recipes-devtools/triton/triton-client/0001-fix-cmake-build.patch:247]: Same as above

**Review by ichergui — CHANGES_REQUESTED** [review]

**Comment by ichergui** [comment]
@roger652 Thanks for the PR 
Overall looks good. please check my comments 
I think we don't have to change `cmake_minimum_required` because master branch provide a recent version anyway.

**Comment by roger652** [comment]
> @roger652 Thanks for the PR Overall looks good. please check my comments I think we don't have to change `cmake_minimum_required` because master branch provide a recent version anyway.

@ichergui You are right. The CMake minimum version only needs to be adjusted for the `dunfell` branch.
I removed all CMake version changes from the `triton-client` recipe.

File: recipes-devtools/triton/triton-python-backend/0001-fix-cmake-build.patch [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e1ea996..8d374d2 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -24,9 +24,9 @@
+ # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ 
+-cmake_minimum_required(VERSION 3.17)
++cmake_minimum_required(VERSION 3.16)
```
Comment (ichergui) [inline recipes-devtools/triton/triton-python-backend/0001-fix-cmake-build.patch:22]: @roger652 Could you please remove this line ?

**Review by ichergui — CHANGES_REQUESTED** [review]
I have a minor comment. Please fix it and I will approve the PR 
Thanks a lot @roger652  

I will may update the `SOURCE_DATE_EPOCH` in `triton_client` recipe. 
I will add you in PR review once I have one.

**Review by ichergui — APPROVED** [review]

**Pull request merged**